tutorials/dataframe/df106_HiggsToFourLeptons.C

```diff
@@ -0,0 +1,281 @@
+         // Reweighting of the samples is different for "data" and "MC"
+         .DefinePerSample("reweighting",
+                          [](unsigned int slot, const ROOT::RDF::RSampleInfo &id) { return id.Contains("mc"); });
+
+   // Define the weight column (scale factor) for the MC samples
+   auto df_mc = df_analysis.Filter("reweighting == true")
+                   .Define("weight", ("scaleFactor_ELE * scaleFactor_MUON * scaleFactor_LepTRIGGER * "
+                                      "scaleFactor_PILEUP * mcWeight * scale * xsecs / sumws * lumi"));
+
+   // Book histograms for individual MC samples
+   auto df_higgs = df_mc.Filter("sample_category == \"higgs\"")
+                      .Histo1D<float>(ROOT::RDF::TH1DModel("higgs", "m4l", 24, 80, 170), "m4l", "weight");
+   auto df_zz = df_mc.Filter("sample_category == \"zz\"")
+                   .Histo1D<float>(ROOT::RDF::TH1DModel("zz", "m4l", 24, 80, 170), "m4l", "weight");
+   auto df_other = df_mc.Filter("sample_category == \"other\"")
+                      .Histo1D<float>(ROOT::RDF::TH1DModel("other", "m4l", 24, 80, 170), "m4l", "weight");
+
+   // Book the invariant mass histogram for the data
+   auto df_h_mass_data = df_analysis.Filter("reweighting == false")
+                            .Filter("sample_category == \"data\"")
+                            .Define("weight_", []() { return 1; })
+                            .Histo1D<float>(ROOT::RDF::TH1DModel("data", "m4l", 24, 80, 170), "m4l", "weight_");
+
+   // Evaluate the systematic uncertainty
+
+   // The systematic uncertainty in this analysis is the MC scale factor uncertainty that depends on lepton
+   // kinematics such as pT or pseudorapidity.
+   // Muons uncertainties are negligible, as stated in https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/PAPERS/MUON
+   // Electrons uncertainties are evaluated based on the plots available in https://doi.org/10.48550/arXiv.1908.0
+   // The uncertainties are linearly interpolated to cover a range of pT values covered by the analysis.
+
+   const std::vector<double> &x = {5.50 * 10e2,  5.52 * 10e2,  12.54 * 10e2, 17.43 * 10e2,
+                                   22.40 * 10e2, 27.48 * 10e2, 30 * 10e2,    10000 * 10e2};
+   const std::vector<double> &y = {0.06628, 0.06395, 0.06396, 0.03372, 0.02441, 0.01403, 0, 0};
+   unsigned int N = x.size();
+   ROOT::Math::Interpolator inter(N, ROOT::Math::Interpolation::kLINEAR);
+   inter.SetData(x, y);
+
+   //  Use the Vary method to add the systematic variations to the total MC scale factor ("weight") of the analysis
+   //  The input consists of the input column to be varied and the lambda function to compute the systematic variations.
+   //  The new output columns contain the varied values of the input column.
+   auto df_with_variations_mc =
+      df_mc
+         .Vary("weight",
+               [&inter](double x, RVec<float> &pt, RVec<unsigned int> &type) {
```

Comment:
```suggestion
               [&inter](double x, const RVecF &pt, RVec<unsigned int> &type) {
```
You can also use your `RVecF` alias to be consistent with how you implemented the main analysis. Also, the `const` should be added here.